idaes/power_generation/unit_models/soec_design.py

```diff
@@ -0,0 +1,621 @@
+#################################################################################
+# The Institute for the Design of Advanced Energy Systems Integrated Platform
+# Framework (IDAES IP) was produced under the DOE Institute for the
+# Design of Advanced Energy Systems (IDAES), and is copyright (c) 2018-2021
+# by the software owners: The Regents of the University of California, through
+# Lawrence Berkeley National Laboratory,  National Technology & Engineering
+# Solutions of Sandia, LLC, Carnegie Mellon University, West Virginia University
+# Research Corporation, et al.  All rights reserved.
+#
+# Please see the files COPYRIGHT.md and LICENSE.md for full copyright and
+# license information.
+#################################################################################
+
+__author__ = "John Eslick"
+
+import enum
+
+import pyomo.environ as pyo
+from pyomo.network import Arc
+from pyomo.common.config import ConfigBlock, ConfigValue, In, Bool
+
+import idaes.core.util.constants as iconst
+from idaes.core import UnitModelBlockData, declare_process_block_class
+from idaes.core.util.config import is_physical_parameter_block
+from idaes.generic_models.properties.core.generic.generic_property import (
+    GenericParameterBlock)
+from idaes.generic_models.properties.core.generic.generic_reaction import (
+    GenericReactionParameterBlock)
+import idaes.generic_models.unit_models as um # um = unit models
+from idaes.core.util.initialization import propagate_state
+import idaes.core.util.scaling as iscale
+import idaes.core.util as iutil
+from idaes.core.util.misc import get_solver
+from idaes.power_generation.properties.natural_gas_PR import (
+    get_prop, get_rxn, EosType)
+from idaes.core.util.exceptions import ConfigurationError, InitializationError
+import idaes.logger as idaeslog
+
+
+@declare_process_block_class(
+    "SoecDesign", doc="Simple SOEC model for process design.")
+class SoecDesignData(UnitModelBlockData):
+    """Simple 0D SOEC model. This model can be used to develop a flowsheet at the
```

Comment:
I have been asking my self if "design" is the best name for this, but I am not sure of a better alternative. Looking at the core libraries for inspiration, I would have said this would be just `SOEC`, and then the more detailed models would derive from this (as appropriate) and specify more details in their names,

Reply:
For a real 0D surrogate, we'd want to have some equation that touches on the voltage-current relationship, and at present this model doesn't have that.

Reply:
@dallan-keylogic, that's right.  You'd have to add that on.  You could inherit this.  But that's what I mean about the 0D name.  I don't want to imply that is more than just a simple design point calculation.